sourmash/lca/lca_utils.py

```diff
@@ -340,23 +354,22 @@ def get_lineage_assignments(self, hashval):
 
         return x
 
-    def _create_signatures(self):
+    def _calc_signatures(self):
         "Create a _signatures member dictionary that contains {idx: minhash}."
         from .. import MinHash
 
-        if not hasattr(self, '_signatures'):
-            minhash = MinHash(n=0, ksize=self.ksize, scaled=self.scaled)
-
-            debug('creating signatures for LCA DB...')
-            sigd = defaultdict(minhash.copy_and_clear)
+        minhash = MinHash(n=0, ksize=self.ksize, scaled=self.scaled)
 
-            for (k, v) in self.hashval_to_idx.items():
-                for vv in v:
-                    sigd[vv].add_hash(k)
+        debug('creating signatures for LCA DB...')
+        sigd = defaultdict(minhash.copy_and_clear)
 
-            self._signatures = sigd
+        for (k, v) in self.hashval_to_idx.items():
+            for vv in v:
+                sigd[vv].add_hash(k)
 
-        debug('=> {} signatures!', len(self._signatures))
+        debug('=> {} signatures!', len(sigd))
+        return sigd
+    _signatures = cached_property(_calc_signatures)
```

Comment:
Why not put this as the decorator on top of the signature?

```
@cached_property
def _signatures(self):
```
instead of `def _calc_signatures(self):`